src/vs/editor/common/config/editorOptions.ts

```diff
@@ -3583,6 +3589,10 @@ export const EditorOptions = {
 			description: nls.localize('peekWidgetDefaultFocus', "Controls whether to focus the inline editor or the tree in the peek widget.")
 		}
 	)),
+	mouseOpensDefinitionInPeek: register(new EditorBooleanOption(
```

Comment:
Overall looking quite good. Not yet sold on the name, maybe have `definitionLink` or `symbolLink` with value set `peek | goto` ("goto" isn't the best option because it might peek) or for a boolean value something like `definitionLinkOpensInPeek`?

Reply:
Yeah, I created a config yesterday with values like `peek | goto` but since it's not always `goto` (it might peek) I decided to create a boolean.

I liked your idea to rename it to `definitionLinkOpensInPeek`.